compiler/rustc_infer/src/infer/freshen.rs

```diff
@@ -146,19 +143,22 @@ impl<'a, 'tcx> TypeFolder<TyCtxt<'tcx>> for TypeFreshener<'a, 'tcx> {
     fn fold_const(&mut self, ct: ty::Const<'tcx>) -> ty::Const<'tcx> {
         match ct.kind() {
             ty::ConstKind::Infer(ty::InferConst::Var(v)) => {
-                let opt_ct =
-                    self.infcx.inner.borrow_mut().const_unification_table().probe_value(v).known();
-                self.freshen_const(opt_ct, ty::InferConst::Var(v), ty::InferConst::Fresh, ct.ty())
+                let mut inner = self.infcx.inner.borrow_mut();
+                let input =
+                    inner.const_unification_table().probe_value(v).known().ok_or_else(|| {
```

Comment:
Can you use `InferCtxt::root_const_var`?

Reply:
that's annoying because the `inner` is already needed anyways for the `probe` :thinking: would merge this as is. We're going to remove the freshener in the medium term anyways

Reply:
Right, I guess I didn't see that probe was being used here too.